cap-primitives/src/std/fs/open_manually.rs

```diff
@@ -201,3 +197,26 @@ fn escape_attempt() -> io::Result<fs::File> {
         "a path led outside of the filesystem",
     ))
 }
+
+#[cold]
+fn symlink_disallowed() -> io::Result<fs::File> {
+    Err(OpenUncheckedError::SymlinkDisallowed.into())
+}
+
+#[derive(Debug)]
+pub(crate) enum OpenUncheckedError {
+    Io(io::Error),
+    SymlinkDisallowed,
```

Comment:
Looking at this more closely, the code returns `SymlinkDisallowed` isn't the code that checks whether they're allowed or not. That check happens at the next level up. So I think we can rename this to just `Symlink` or `SymlinkError`.